src/commands/summon.rs

```diff
@@ -1,31 +1,36 @@
 use std::time::Duration;
 
+use serenity::client::Context;
+use serenity::model::interactions::application_command::ApplicationCommandInteraction;
 use serenity::prelude::Mentionable;
-use serenity::{
-    client::Context,
-    framework::standard::{macros::command, CommandResult},
-    model::channel::Message,
-};
+use serenity::prelude::SerenityError;
 use songbird::Event;
 
 use crate::events::idle_handler::IdleHandler;
 use crate::strings::AUTHOR_NOT_FOUND;
-use crate::utils::send_simple_message;
+use crate::utils::create_response;
+
+pub async fn summon(
+    ctx: &Context,
+    interaction: &mut ApplicationCommandInteraction,
+    send_reply: bool,
+) -> Result<(), SerenityError> {
+    let guild_id = interaction.guild_id.unwrap();
+    let guild = ctx.cache.guild(guild_id).await.unwrap();
 
-#[command]
-#[aliases("join")]
-pub async fn summon(ctx: &Context, msg: &Message) -> CommandResult {
-    let guild = msg.guild(&ctx.cache).await.unwrap();
     let manager = songbird::get(ctx).await.unwrap();
 
     let channel_opt = guild
         .voice_states
-        .get(&msg.author.id)
+        .get(&interaction.user.id)
         .and_then(|voice_state| voice_state.channel_id);
 
     let channel_id = match channel_opt {
         Some(channel_id) => channel_id,
-        None => return send_simple_message(&ctx.http, msg, AUTHOR_NOT_FOUND).await,
+        None if send_reply => {
+            return create_response(&ctx.http, interaction, AUTHOR_NOT_FOUND).await
+        }
```

Comment:
Does cargo fmt force these {} ?

Reply:
Yeah, to distinguish an assignment from a function exit.
If I do it inline, Rust complains that it expected a `ChannelId` but got a `Result<(), SerenityError>`.